src/groupByKey.ts

```diff
@@ -1,7 +1,13 @@
 type GroupsMap<T> = {
-  [key: string]: T[];
+  [key: string | number]: T[];
 };
 
-export function groupByKey(items, key) {
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
   // write code here;
+  return items.reduce((acc: GroupsMap<T>, item: T) => {
+    const value = item[key] as string | number;
+    const sortedItem = items.filter((el) => el[key] === value);
+
+    return { ...acc, [value]: sortedItem };
```

Comment:
The return statement here is incorrect. You should not create a new object with `{ ...acc, [value]: sortedItem }` for each iteration. Instead, modify the accumulator `acc` directly by adding the current `item` to the array corresponding to `value`.